### ISSUE TITLE
The issue with files extension for ABC.dwg (AutoCAD) file

### ISSUE THREAD

**Issue by pro2on** [anggrayudi/SimpleStorage#67]
Library version: 0.13.0
OS version: [Android 11]
Device model: [Samsung S10+]

**Describe the bug**
While copy files with DWG extension the library adds the BIN extension to the end, so
ABC.dwg -> ABC.dwg.bin.

As a result it's impossible to open this file after copy with DWG viewer.

**To Reproduce**
Run Sample application and copy any DWG file (you can find example in google) from one folder to another one.

**Stacktrace**
The issue is in MimeType.getFullFileName method. Why you are forcing to use some additional BIN extension instead of using original filename?

**Comment by pro2on** [comment]
possible fix, but not sure if it's correct for all cases:

```
    /**
     * * Given `name` = `ABC` AND `mimeType` = `video/mp4`, then return `ABC.mp4`
     * * Given `name` = `ABC` AND `mimeType` = `null`, then return `ABC`
     * * Given `name` = `ABC.mp4` AND `mimeType` = `video/mp4`, then return `ABC.mp4`
     * @param name can have file extension or not
     */
    @JvmStatic
    fun getFullFileName(name: String, mimeType: String?): String {
        return if (getExtensionFromFileName(name).isEmpty()) {
            // Prior to API 29, MimeType.BINARY_FILE has no file extension
            getExtensionFromMimeType(mimeType).let {
                if (it.isEmpty() || name.endsWith(".$it")) name else "$name.$it".trimEnd('.')
            }
        } else {
            name
        }
    }
    
    @JvmStatic
    fun getExtensionFromFileName(filename: String?): String {
        return filename?.substringAfterLast('.', "") ?: ""
    }

    /**
     * @see getExtensionFromMimeType
     */
    @JvmStatic
    fun getExtensionFromMimeTypeOrFileName(mimeType: String?, filename: String): String {
        return if (mimeType == null || mimeType == UNKNOWN) getExtensionFromFileName(filename) else getExtensionFromMimeType(mimeType)
    }
```

**Comment by pro2on** [comment]
was opened PR

**Comment by anggrayudi** [comment]
Thanks for your help. I have commented your PR, please check. 👍

**Comment by anggrayudi** [comment]
Fixed in [0.14.0](https://github.com/anggrayudi/SimpleStorage/releases/tag/0.14.0)
